hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -3469,12 +3469,18 @@ private void scan(HBaseRpcController controller, ScanRequest request, RegionScan
           limitReached = sizeLimitReached || timeLimitReached || resultsLimitReached;
```

Comment:
Not in the scope of this backport ticket, but why do we check only those two limits ?
is batch size not applicable here ? Why do we ignore the heap size limit ?

Maybe worth looking into this in another JIRA.